app/models/registration.ts

```diff
@@ -84,6 +81,27 @@ export default class RegistrationModel extends NodeModel.extend() {
 
     @hasMany('institution', { inverse: 'registrations' })
     affiliatedInstitutions!: DS.PromiseManyArray<InstitutionModel> | InstitutionModel[];
+
+    registrationStateMap(): any {
```

Comment:
```suggestion
    registrationStateMap(): Record<RegistrationState, boolean> {
```

Reply:
Can you explain this?

Reply:
`Record<K, V>` describes an object with keys of type `K` and values of type `V`. It's useful shorthand for the more arcane `{ [Key in K]: V }`.

Reply:
[`Record` reference](https://www.typescriptlang.org/docs/handbook/advanced-types.html#mapped-types)

Reply:
I guess the only thing here is that it does Record<string, boolean> rather than RegistrationState.

Reply:
Since `RegistrationState` is a string enum, it's sort of a subset of the `string` type.
```
let foo: RegistrationState;
```
is the same as
```
let foo: 'Public' | 'Embargoed' | 'Withdrawn' | ...;
```
`Record<string, boolean>` would be fine too, but it's usually good to try for the most specific type we reasonably can.